test/js-native-api/common.h

```diff
@@ -58,3 +58,24 @@
 
 #define DECLARE_NAPI_GETTER(name, func)                                  \
   { (name), NULL, NULL, (func), NULL, NULL, napi_default, NULL }
+
+#define NAPI_STATUS_TO_STRING(status)                                         \
+  (status) == napi_ok ? "napi_ok" :                                           \
+  (status) == napi_invalid_arg ? "napi_invalid_arg" :                         \
```

Comment:
I'm not sure the assigning of status is required here and I think it could be removed on this and the following lines. If you check the preprocessor output it looks like this at the moment:
```console
$ clang -Isrc -E test/js-native-api/test_constructor/test_constructor.c
...
do { if ((napi_create_string_utf8(env, (ret[0]) == napi_ok ? "napi_ok" : (ret[0]) == napi_invalid_arg ? "napi_invalid_arg" : (ret[0]) == napi_object_expected ? "napi_object_expected" : (ret[0]) == napi_string_expected ? "napi_string_expected" : (ret[0]) ==      napi_name_expected ? "napi_name_expected" : (ret[0]) == napi_function_expected ? "napi_function_expected" : (ret[0]) == napi_num     ber_expected ? "napi_number_expected" : (ret[0]) == napi_boolean_expected ? "napi_boolean_expected" : (ret[0]) == napi_array_expe     cted ? "napi_array_expected" : (ret[0]) == napi_generic_failure ? "napi_generic_failure" : (ret[0]) == napi_pending_exception ? "     napi_pending_exception" : (ret[0]) == napi_cancelled ? "napi_cancelled" : (ret[0]) == napi_escape_called_twice ? "napi_escape_cal     led_twice" : (ret[0]) == napi_handle_scope_mismatch ? "napi_handle_scope_mismatch" : (ret[0]) == napi_callback_scope_mismatch ? "     napi_callback_scope_mismatch" : (ret[0]) == napi_queue_full ? "napi_queue_full" : (ret[0]) == napi_closing ? "napi_closing" : (re     t[0]) == napi_bigint_expected ? "napi_bigint_expected" : (ret[0]) == napi_date_expected ? "napi_date_expected" : "UNKNOWN_ERROR",      18446744073709551615UL, &prop_value)) != napi_ok) { do { const napi_extended_error_info *error_info; napi_get_last_error_info(((     env)), &error_info); _Bool is_pending; napi_is_exception_pending(((env)), &is_pending); if (!is_pending) { const char* error_mess     age = error_info->error_message != ((void*)0) ? error_info->error_message : "empty error message"; napi_throw_error(((env)), ((vo     id*)0), error_message); } } while (0); return ((void*)0); } } while (0);
```

This might just be my own preference but I think I would prefer this as a c function that uses a macro to create the enum strings. Something like:
```c
#define FOREACH_STATUS(STATUS)                   \
        STATUS(napi_ok)                          \
        STATUS(napi_invalid_arg)                 \
        STATUS(napi_object_expected)             \
        STATUS(napi_string_expected)             \
        STATUS(napi_name_expected)               \
        STATUS(napi_function_expected)           \
        STATUS(napi_number_expected)             \
        STATUS(napi_boolean_expected)            \
        STATUS(napi_array_expected)              \
        STATUS(napi_generic_failure)             \
        STATUS(napi_pending_exception)           \
        STATUS(napi_cancelled)                   \
        STATUS(napi_escape_called_twice)         \
        STATUS(napi_handle_scope_mismatch)       \
        STATUS(napi_callback_scope_mismatch)     \
        STATUS(napi_queue_full)                  \
        STATUS(napi_closing)                     \
        STATUS(napi_bigint_expected)             \
        STATUS(napi_date_expected)


#define GENERATE_STRING(STRING) #STRING,

static const char* status_strings[] = {
    FOREACH_STATUS(GENERATE_STRING)
};
```
From this the preprocess will produce the following:
```c
$ clang -Isrc -Itest/js-native-api -E test/js-native-api/common.h | grep -A 3 status_strings
static const char* status_strings[] = {
    "napi_ok", "napi_invalid_arg", "napi_object_expected", "napi_string_expected", "napi_name_expected", "napi_function_expected", "napi_number_expected", "napi_boolean_expected", "napi_array_expected", "napi_generic_failure", "napi_pending_exception", "napi_cancelled", "napi_escape_called_twice", "napi_handle_scope_mismatch", "napi_callback_scope_mismatch", "napi_queue_full", "napi_closing", "napi_bigint_expected", "napi_date_expected",
};
```
And a function named `napi_status_to_string` could be added to `test_constructor.c`:
```c
static const char* napi_status_to_string(napi_status status) {
  return status_strings[status];
}
```
And the code in the test would in this case look like:
```c
NAPI_CALL(env, napi_create_string_utf8(env,
                                         napi_status_to_string(ret[5]),
                                         NAPI_AUTO_LENGTH,
                                         &prop_value));
```